ios/LayoutReanimation/REAAnimationsManager.m

```diff
@@ -119,47 +119,56 @@ - (void)findRoot:(UIView *)view roots:(NSMutableSet<NSNumber *> *)roots
   }
 }
 
-- (BOOL)dfs:(UIView *)root view:(UIView *)view cands:(NSMutableSet<NSNumber *> *)cands
+- (BOOL)dfs:(UIView *)root view:(UIView *)view
 {
   NSNumber *tag = view.reactTag;
   if (tag == nil) {
     return true;
   }
-  if (![cands containsObject:tag] && _states[tag] != nil) {
+  if (_viewsToRemove[tag] == nil && _states[tag] != nil) {
     return true;
   }
   BOOL cannotStripe = false;
   NSArray<UIView *> *toRemoveCopy = [view.reactSubviews copy];
   for (UIView *child in toRemoveCopy) {
     if (![view isKindOfClass:[RCTTextView class]]) {
-      cannotStripe |= [self dfs:root view:child cands:cands];
+      cannotStripe |= [self dfs:root view:child];
     }
   }
   if (!cannotStripe) {
     if (view.reactSuperview != nil) {
       [_reaUiManager unregisterView:view];
     }
     [_states removeObjectForKey:tag];
-    [_viewForTag removeObjectForKey:tag];
-    [_toRemove removeObject:tag];
+    [_viewsToRemove removeObjectForKey:tag];
+    [_viewsWithExitingAnimation removeObjectForKey:tag];
```

Comment:
It still seem like there are cases where views can be kept in those dicts. Specifically, it appears like the only codepath that leads to the removal is when the given view is returned from `findRoot`. But `findRoot` can in some cases return the parent of a given view and not the view itself. It'd feel safer if we have a cleaner contract as to when given view needs to be removed from an array. Also it'd be good to add a few lines of documentation for each of these two new dicts. I can understand the purpose of `viewWithExitingAnimations` fine by its name but not so much with` viewToRemove` – i.e. I don't know why we need to keep track of all views we remove.

As to "a cleaner contract" an example would be to always remove object from `viewWithExitingAnimations` after animation is finished (have a callback of some sort ?). With `viewsToRemove` we should ideally have something similar. Now it is very difficult to track the codepath that leads to deletion and it can be altered to easily.

Reply:
yes, in the perf monitor i can see that the total number of rendered view increase over time. 
My app's lifecycle is: 
Mount 1st screen -> navigate to 2nd screen -> navigate to 3rd screen -> navigate to last screen -> Reset navigation to 1st screen and then start again the same cycle. 

Previously I've seen custom UI views not being unmounted but this PR fixes that issue. But i can see an increase of total number of rendered views with each new session.